products/dns/terraform.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Provider::ResourceOverrides
+  ManagedZone: !ruby/object:Provider::Terraform::ResourceOverride
+    id_format: "{{name}}"
+    example:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "dns_managed_zone_basic"
+        primary_resource_id: "prod"
+        version: <%= _version_name %>
+        vars:
+          zone_name: "prod-zone"
+    properties:
+      creationTime: !ruby/object:Provider::Terraform::PropertyOverride
+        exclude: true
+      description: !ruby/object:Provider::Terraform::PropertyOverride
+        description: |
+          A textual description field. Defaults to 'Managed by Terraform'.
+        default_value: 'Managed by Terraform'
+      dnsName: !ruby/object:Provider::Terraform::PropertyOverride
+        required: true
```

Comment:
Can this be marked required in api.yaml instead? That's more of an API thing vs a TF thing.